### PR TITLE
style: tweak style of code titles

### PR DESCRIPTION
**Pull request by DSchau**
This PR improves the style of code titles per @fk :)

High-level changes:

- Reduce font size
- Reduce padding
- Co-locate some shared code styling (e.g. font family, font size, ligatures, etc.) with other code blocks

Additionally, I _could_ modify gatsby-remark-code-titles to use a nested code block for the title, but this works for now.

screen shot 2018-09-06 at 10 09 41 am

### CODE REVIEW

**Comment by fk** [comment]
> Additionally, I could modify gatsby-remark-code-titles to use a nested code block for the title, but this works for now.

I think that's not worth the effort…might even be not-so-nice in terms of semantics?
I have to admit I grew lazy on those a bit… 😁 but I'd interpret path and filename more as a title than code here? (I'm blushing a little trying to figure out when was the last time I looked up stuff in the spec :))

**Comment by KyleAMathews** [comment]
How does this look on a long page? It seems a bit _too_ purple so it'd be overly attention grabbing perhaps.

**Comment by KyleAMathews** [comment]
Could make sense to just have a thin purple border or something that's less attention grabbing.

**Comment by fk** [comment]
I totally forgot this, but I actually threw together the elements of a "docs" page in [Figma](https://www.figma.com/file/eY2XGFAV06IW8CG3xGNSYwzY/gatsbyjs.org?node-id=78%3A4660) a couple of days ago (disclaimer: not a final design; not even the paragraph styles are consistent ;-)).

Apparently I also added a code block containing path + filename (and a copy button). It's as undesigned and unfinished as the whole page is, but it's close enough to what Kyle suggests:

image

(Didn't manage to disable the ligatures of Space Mono in Figma yet ;-))

**Comment by fk** [comment]
Bottom border (highlighted line colors now matching the current ones):

image


image

File: www/src/utils/typography.js [hunk 1]
```diff
@@ -297,11 +297,6 @@ const _options = {
             ((options.blockMarginBottom * 1.5) / 5) * 1
           )}`,
         },
-        ".gatsby-code-title": {
-          padding: `${rhythm(options.blockMarginBottom)} ${rhythm(
-            options.blockMarginBottom * 1.5
-          )}`,
-        },
```
[the remark this answers is not in the document]
Comment (DSchau) [reply www/src/utils/typography.js]: Silly mistake 😅 Will address!

**Comment by DSchau** [comment]
@fk those mockups are _great_. What if we go with 2 or 3 (I'll modify this PR) and then tweak as needed? I like how it's quite a bit more subtle, and leaves space for the copy button too!

File: www/src/utils/typography.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (fk) [reply www/src/utils/typography.js]: No worries! Also not silly — using [two sets of breakpoints](https://github.com/gatsbyjs/gatsby/issues/7944) is though ;-), or [that](https://github.com/gatsbyjs/gatsby/issues/6889) 🙄

**Comment by fk** [comment]
Looking at those again, I'm leaning towards 3

image

because that conflicts the least with potentially highlighted lines of code?!

File: www/src/utils/typography.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (DSchau) [reply www/src/utils/typography.js]: @fk I went with 3, and then did some slight refactors. I'll comment below!

**Comment by DSchau** [comment]
@fk here's where I ended up.

1. Removed the border radius entirely (I think this looks cleaner anyways?)
2. Used the same border color as the boxShadow on the gatsby-code-highlight element
    - Note: I used `border` instead of box-shadow so I can remove the bottom border in an effort to not double up with the box-shadow
3. Moved some of the code snippet colors into src/utils/colors.js in an effort to keep things a little dryer
4. Used a slightly darker code title color than in your mockup (the Chrome color picker was helpful here re: accessibility)

Should be good to go? Feedback welcome, so let me know!

File: www/src/utils/colors.js [hunk 2]
```diff
@@ -78,6 +78,9 @@ const colors = {
     light: gray(80, 270),
   },
   code: {
+    bg: '#fdfaf6',
+    border: '#faede5',
+    text: '#866c5b',
```
Comment (fk) [inline www/src/utils/colors.js]: 💯 🙏

**Comment by fk** [comment]
> Removed the border radius entirely (I think this looks cleaner anyways?)

I was about to write s/th like "let's bring them back (happy to do this myself)—we use them in a couple of other places (buttons, inputs, images) so let's stay consistent for now"…but then I realized it seems we lost them somewhere along the way for images, didn't consistently stick to them for new elements like e.g. the newsletter signup box…so:
Let's get this in radius-free for now (and let me take care of brining them back consistently)!

👍🙏to all other changes!
Thanks so much for making things DRYer and thinking of contrast ratios 🤗 — I have a slight issue with making things to bright and subtle 😸…never failing at that! 🙄

**Comment by DSchau** [comment]
🎉 Thanks for the review @fk :)

**Pull request merged**